curve_apps/utils.py

```diff
@@ -12,17 +12,75 @@
 from __future__ import annotations
 
 import re
+from collections.abc import Callable
 
 import numpy as np
+from geoapps_utils.numerical import weighted_average
 from geoh5py.objects import Curve, Grid2D, ObjectBase, Points, Surface
 from matplotlib.contour import ContourSet
```

Comment:
Can remove this, and also remove matplotlib from the toml

Reply:
done